tests/test_kldiv.py

```diff
@@ -18,225 +18,133 @@
-            self.chol = tf.stack([tf.diag(self.sqrt[:, i]) for i in range(N)])
-
-    def test_white(self):
-        with self.test_session() as sess:
-            kl_diag = gpflow.kullback_leiblers.gauss_kl(self.mu, self.sqrt)
-            kl_dense = gpflow.kullback_leiblers.gauss_kl(self.mu, self.chol)
-
-            res_diag = sess.run(kl_diag, feed_dict=self.feed_dict)
-            res_dense = sess.run(kl_dense, feed_dict=self.feed_dict)
-
-            np.testing.assert_allclose(res_diag, res_dense)
-
-    def test_nonwhite(self):
-        with self.test_session() as sess:
-            kl_diag = gpflow.kullback_leiblers.gauss_kl(self.mu, self.sqrt, self.K)
-            kl_dense = gpflow.kullback_leiblers.gauss_kl(self.mu, self.chol, self.K)
-
-            res_diag = sess.run(kl_diag, feed_dict=self.feed_dict)
-            res_dense = sess.run(kl_dense, feed_dict=self.feed_dict)
+    np.testing.assert_allclose(kl_diag.eval(),
+                               kl_dense.eval())
 
-            np.testing.assert_allclose(res_diag, res_dense)
-
-
-class WhitenedTest(GPflowTestCase):
+@pytest.mark.parametrize('diag', [True, False])
+def test_whitened(session_tf, diag, mu, sqrt_diag, I):
     """
     Check that K=Identity and K=None give same answer
     """
+    chol_from_diag = tf.stack([tf.diag(sqrt_diag[:, i]) for i in range(Datum.N)]) # N x M x M
+    s = sqrt_diag if diag else chol_from_diag
 
-    def setUp(self):
-        with self.test_session():
-            N = 4
-            M = 5
-            self.mu = tf.placeholder(settings.float_type, [M, N])
-            self.sqrt = tf.placeholder(settings.float_type, [M, N])
-            self.chol = tf.placeholder(settings.float_type, [M, M, N])
-            self.I = tf.placeholder(settings.float_type, [M, M])
-
-            self.rng = np.random.RandomState(0)
-            self.mu_data = self.rng.randn(M, N)
-            self.sqrt_data = self.rng.randn(M, N)
-            q_sqrt = np.rollaxis(np.array([np.tril(self.rng.randn(M, M)) for _ in range(N)]),
-                                 0, 3)
-            self.chol_data = q_sqrt
-
-            self.feed_dict = {
-                self.mu: self.mu_data,
-                self.sqrt: self.sqrt_data,
-                self.chol: self.chol_data,
-                self.I: np.eye(M),
-            }
-
-    def test_diag(self):
-        with self.test_session() as sess:
-            kl_white = gpflow.kullback_leiblers.gauss_kl(self.mu, self.sqrt)
-            kl_nonwhite = gpflow.kullback_leiblers.gauss_kl(self.mu, self.sqrt, self.I)
-
-            res_white = sess.run(kl_white, feed_dict=self.feed_dict)
-            res_nonwhite = sess.run(kl_nonwhite, feed_dict=self.feed_dict)
+    kl_white = gauss_kl(mu, s)
+    kl_nonwhite = gauss_kl(mu, s, I)
 
-            np.testing.assert_allclose(res_white, res_nonwhite)
+    np.testing.assert_allclose(kl_white.eval(),
+                               kl_nonwhite.eval())
```

Comment:
I apologize for the pickiness, but have lines longer than 80 chars and here you injected newline. Why is that?